### PR TITLE
Adds information on CASINO

### PR DESCRIPTION
**Pull request by aturner-epcc**

### CODE REVIEW

File: docs/other-software/casino.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+```
+module load PrgEnv-gnu
+module load craype-network-ucx
+module load cray-mpich-ucx
+```
+
+### Setting the number of cores sharing memory
+
+In your job submission script you set the number of cores sharing memory segements
```
Comment (juanfrh) [inline docs/other-software/casino.md:62]: _Segments_ is mispelled.

File: docs/other-software/casino.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  the best performing choices. If memory efficiency is critical, then 32 cores gives
+  good performance. More than 32 cores sharing a memory segment gives poor performance.
+- Ensuring that MPI processes are pinned to cores in a sequential manner so cores that 
+  are sharing shared memory segments are located as close to each other as possible.
+
+Next, we show how to make sure that the MPI transport layer is set to UCX, 
+how to set the number of cores sharing the System-V shared memory segments and
+how to pin MPI processes sequentially to cores.
+
+Finally, we provide an example CASINO job submission script that demonstrates all
```
Comment (juanfrh) [inline docs/other-software/casino.md:45]: I find "an example CASINO job submission script" quite long to read. For instance, I'd use "a job submission script".

File: docs/other-software/casino.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+### CASINO_NUMABLK=8
+
+| Nodes | Time taken (s) |
+|------:|---------------:|
+|     1 |         289.90 |
+|     2 |         154.93 |
+|     4 |          81.06 |
+|     8 |          41.44 |
+|    16 |          23.16 |
```
Comment (juanfrh) [inline docs/other-software/casino.md:144]: Maybe adding the speedup (or efficiency) would make the performance results easier to understand. It's just a thought.

**Review by juanfrh — CHANGES_REQUESTED** [review]
Thanks Andy, it looks great. I've found a typo. I've also added two suggestions, feel free to address/scrap them.

File: docs/other-software/casino.md (hunk 1 above)
Comment (aturner-epcc) [reply docs/other-software/casino.md]: Well spotted, fixed.

File: docs/other-software/casino.md (hunk 2 above)
Comment (aturner-epcc) [reply docs/other-software/casino.md]: Changed

File: docs/other-software/casino.md (hunk 3 above)
Comment (aturner-epcc) [reply docs/other-software/casino.md]: Speedup added

**Comment by aturner-epcc** [comment]
Thanks @juanfrh I have made the suggested changes

**Review by juanfrh — APPROVED** [review]
Looks great, thanks a lot.

**Pull request merged**